src/main/java/org/terasology/logic/market/PurchasableComponent.java

```diff
@@ -0,0 +1,23 @@
+/*
+ * Copyright 2019 MovingBlocks
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.terasology.logic.market;
+
+import org.terasology.entitySystem.Component;
+import org.terasology.world.block.items.AddToBlockBasedItem;
+
+@AddToBlockBasedItem
+public class PurchasableComponent implements Component {
```

Comment:
Is this supposed to contain more information later on, or is it just a marker component? Due to the annotation `@AddToBlockBasedItem`, is this for generic purchasable blocks and items, or just for blocks?

Reply:
Yes, it will contain information about what should be the 'purchasable item' purchased against. This component is for both blocks as well as prefab items. I am sceptical about the use-case of this annotation too. This annotation is used when you want the blocks to be detected with some component, right?